### PR TITLE
Add Makefile target postflight-cli

### PR DESCRIPTION
**Pull request by branden**
The `postflight-cli` target runs the installation postflight check using the installer's CLI.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -134,6 +134,9 @@ start: build clean-certs $(CERTS_DIR) clean-containers master agent public_agent
 postflight: ## Polls DC/OS until it is healthy (5m timeout)
 	@docker exec $(INTERACTIVE) $(MASTER_CTR)1 dcos-postflight
 
+postflight-cli: ## Runs the DC/OS installer with --postflight.
+	@bash $(DCOS_GENERATE_CONFIG_PATH) --postflight --offline -v
```
Comment (karlkfi) [inline Makefile]: Your fork is out of date. You'll want:
```
@echo "+ Running postflight"
$(INSTALLER_CMD) --postflight
```

**Comment by karlkfi** [comment]
What version was --postlfight added? I know it's not new.

I can't remember why we've been using a custom postflight script, unless it was some sort of version compatibility to support older versions of DC/OS that didn't have postflight.

It could also have been to validate that the GUI is accessible tho. I think the CLI postflight just checks 3DT state.

Perhaps we need to modify `make postflight` to run `--postflight` and then another script that just pools the GUI?

**Comment by branden** [comment]
@karlkfi I pushed an update that resolves your comment.

`--postflight` has been around as long as the CLI installer, and at least as long as the web installer. Pre-1.7, at least. The CLI and web installer postflight both verify the same thing: component health via the 3DT health API. One of those components is the master's adminrouter, which we effectively use as a proxy for determining whether the GUI is available. Either postflight approach (CLI or web) should work for the purposes of dcos-docker, but it's nice to have both as options in the Makefile since this is for local development.

IMO if the built-in postflight isn't sufficient for determining whether the cluster is up and available, that's a bug in DC/OS. Now that checks are becoming a thing and postflight is going to using them, we can add a GUI check there.

**Comment by karlkfi** [comment]
So... ultimately I do want to use `--postflight` instead of my own script. But as it stands, the existing `--postflight` behavior is  pretty terrible. It regularly prints stack traces, doesn't indicate any sort of progress, and has some odd behavior.

I supposed having a `postflight-cli` is a tolerable placeholder for now, but it would really only be useful for people testing that cli command.

AFAIK, the only things that `--postflight` does that `make postflight` doesn't is:
- check multiple masters
- check all agents

These two behaviors can be trivially added to `make postflight` just by adding some loops. Checking multiple masters we can add right away. Checking agents tho is debatable whether it's actually desirable. The UI & API is usable as soon as the masters are up. So from a user perspective, that's all they need. But from a CI perspective, you'd want to make sure all the agents are up before you start running integration tests or to verify that code changes still allow the agents to come up.

So there are really two classes of use cases for postflight, which means we probably want either two commands or an optional argument on one command...

**Comment by karlkfi** [comment]
So I guess what I'm saying is:
- I don't want people to get attached to this command because i may remove it in the future.
- I think it having two kinds of postflights might confuse people who don't knwo the difference and make UX worse...
- If I add this we need some docs or at least a `make help` comment about why we have two.
- If you still think you need this after my comments, let me know.

What do you think?
What exactly is your use case?

**Comment by branden** [comment]
My use case was that I wanted to test the CLI postflight locally since I was hacking on it, and there was no existing Makefile target which allowed me to do that. It's a limited use case, which is why I didn't try to change the existing postflight target.

I doubt that running postflight on agents would make it take much longer, agents tend to come up quickly once a cluster's Mesos and DNS are working.

Would adding some comments and help output explaining that this is duplicated functionality useful only for local DC/OS development do the trick?

**Comment by karlkfi** [comment]
```
Would adding some comments and help output explaining that this is duplicated functionality useful only for local DC/OS development do the trick?
```

I'd personally rather someone improved the upstream DC/OS postflight check to be as helpful and verbose as the bash script I wrote here. Then I could remove it here and just delegate to the cli command. But I think it would need to be ported from bash to python in order to do so, and my python is pretty rusty...

**Comment by branden** [comment]
This is a small change that I don't mind maintaining myself, so if this isn't useful to the project then I'll go ahead and close this out.

**Pull request closed without merging**